### PR TITLE
cli: add experimental package fix command

### PR DESCRIPTION
**Pull request by Rugvip**
## Hey, I just made a Pull Request!

This is the best option I've found for this so far, having tried making this a direct ESLint rule as well.

Very experimental this one, would like to get a feel for if this is something I and other people would actually use and find useful, or if it just rots and we remove it.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/cli/src/commands/add-deps.ts [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+      rules: {
+        'import/no-extraneous-dependencies': [
+          'error',
+          {
+            devDependencies: [
+              `!${joinPath(paths.targetDir, 'src/**')}`,
+              joinPath(paths.targetDir, 'src/**/*.test.*'),
+              joinPath(paths.targetDir, 'src/**/*.stories.*'),
+              joinPath(paths.targetDir, 'src/setupTests.*'),
+            ],
```
Comment (freben) [inline packages/cli/src/commands/add-deps.ts:63]: Is there a case for adding things from the dev folder too where appropriate?

**Review by freben — APPROVED** [review]
Looks good. I think in terms of longer term usability, I think if it was more along lines of a more general "fixup deps" command that also made sure to fix wrong in-repo versions etc, I'd bind it to a hot key and use it a lot

File: packages/cli/src/commands/add-deps.ts (hunk 1 above)
Comment (Rugvip) [reply packages/cli/src/commands/add-deps.ts:63]: First pattern takes care of that, anything outside src

**Comment by Rugvip** [comment]
@freben yep I like it, prolly worth renaming this to something closer to that

**Comment by freben** [comment]
`versions:arrange`?

**Comment by Rugvip** [comment]
`package fix`

With

```json
"scripts": {
  "fix": "backstage-cli package fix"
}
```

? :>

**Comment by freben** [comment]
0E7BBF78-1B91-4A14-98B3-4859B5796A36

**Comment by freben** [comment]
`also-write-the-code-for-me-in-the-first-place --then-fix`

**Comment by freben** [comment]
That's a big promise to make (pretty broad name of the command) :) Might it have made sense to do `fix:dependencies` and making it a little subfamily of things? But alright!

**Comment by Rugvip** [comment]
Yeah I know, made it `fix --deps` to make it possible to make it more narrow. Very experimental for now and I think we might even end up with a separate command category for this kind of automation. Gonna :shipit: and let's see how it feels

**Pull request merged**